pages/profile/[vin].js

```diff
@@ -1,20 +1,58 @@
+import EditCarForm from "@/components/EditCarForm";
 import Image from "next/image";
 import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
 import Cars from "../../db/db.json";
+
 export default function CarDetails() {
+  const [isEditing, setIsEditing] = useState(false);
   const router = useRouter();
   const { vin } = router.query;
-  const activeCar = Cars.find((car) => {
-    return car.VIN === vin;
-  });
-  console.log(activeCar);
+
+  const [activeCar, setActiveCar] = useState(null);
+
+  useEffect(() => {
+    async function getCar() {
+      const data = Cars.find((car) => {
+        return car.VIN === vin;
+      });
+      setActiveCar(data);
+    }
+    getCar();
+  }, [vin]);
```

Comment:
```suggestion

const currentCar = Cars.find((car) => {
        return car.VIN === vin;
})
 const [activeCar, setActiveCar] = useState(currentCar);

```

This function is neither asynchronous, nor does it require a useEffect.

Reply:
Thanks for the review, i made this change on my local project, but then page displays just "loading..." and does not move on from there. Because of that i implemented the useState. Should i go on with the suggested changes and search for another way, to fix the problem or is it ok to use the useEffect?